### PR TITLE
docs: improve "explicit version" section in readme

### PR DESCRIPTION
**Pull request by MikeMcC399**
This PR re-works the [README section: Explicit version](https://github.com/cypress-io/github-action/blob/master/README.md#explicit-version).

The text is simplified, corrected and the heading is renamed using the more neutral "Action version" instead of "Explicit version".

- `v5` is clarified as being the latest version
- Selecting `cypress-io/github-action@v5` is using a branch name, not a tag name. (There is no tag `v5`.)
- In addition to referring to GitHub's [tags](https://github.com/cypress-io/github-action/tags) list, the [releases](https://github.com/cypress-io/github-action/releases) list and [Changelog](https://github.com/cypress-io/github-action/blob/master/README.md#changelog) are also linked

### CODE REVIEW

**Comment by MikeMcC399** [comment]
The text is conveniently reviewable under https://github.com/MikeMcC399/github-action/blob/docs/version-selection/README.md#action-version.

Once this section is stable I would submit a follow-on PR to align https://docs.cypress.io/guides/continuous-integration/github-actions#Explicit-Version-Number with the repo README text.

**Review by AtofStryker — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -102,25 +102,27 @@ jobs:
...
+
+Alternatively, to mitigate unforeseen breaks, bind to a specific [tag](https://github.com/cypress-io/github-action/tags), for example:
 
 ```yml
 - name: Cypress run
   uses: cypress-io/github-action@v5.1.0
 ```
 
-By using the full version tag, you will avoid accidentally using a newer version of the action.
+The changes associated with each tag are shown under GitHub's [releases](https://github.com/cypress-io/github-action/releases) list. Refer also to the [Changelog](#changelog) for an overview of major changes.
```
Comment (emilyrohrbough) [inline README.md:125]:
```suggestion
The changes associated with each tag are shown under GitHub's [releases](https://github.com/cypress-io/github-action/releases) list. Refer also to the [Changelog](#changelog) for an overview of release changes.
```
are we only documented breaking changes?

Comment (MikeMcC399) [reply README.md:125]: @emilyrohrbough

Thanks for your review!

The repository uses a default setup of [cycjimmy/semantic-release-action](https://github.com/cycjimmy/semantic-release-action) to produce releases. This is documented on the standard GitHub [releases](https://github.com/cypress-io/github-action/releases) page. Also there is another open PR https://github.com/cypress-io/github-action/pull/853 which goes into more detail about the current release process, which you might find helpful.

There is no automatic Changelog produced. There is a pending draft PR https://github.com/cypress-io/github-action/pull/752 to activate the production of a changelog, which is waiting for @jaffrepaul to have time on improving the release workflow. To begin with, @jaffrepaul was editing the individual release entries on the [releases](https://github.com/cypress-io/github-action/releases) page to include more detail, including commits which were not triggering a release, but lately he has stopped doing this. Presumably due to lack of time.

Because there was no overview of any releases, let alone major releases, and the existing migration guide was unhelpful, I added an overview of major releases in PR https://github.com/cypress-io/github-action/pull/769. This is just a short table included in the [README: Changelog](https://github.com/cypress-io/github-action/blob/master/README.md#changelog).

So, in summary, the original text is correct and your suggested change would be inaccurate for this repository.

I'm happy to work together if there is any work done to improve the release workflow, however I do appreciate your heavy workload and I'm not expecting anything soon on this.

**Comment by MikeMcC399** [comment]
- PR set to draft pending merge of https://github.com/cypress-io/github-action/pull/855

File: README.md (hunk 1 above)
Comment (emilyrohrbough) [reply README.md:125]: @MikeMcC399 Wow! Thank you for the full context!! I am just ramping up with this project and where it is at. @jaffrepaul Recently shifted teams and focus at Cypress and likely won't be actively/frequently contributing to this project due to his shift in focus.

**Review by emilyrohrbough — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (MikeMcC399) [reply README.md:125]: @emilyrohrbough

> Wow! Thank you for the full context!! I am just ramping up with this project and where it is at. @jaffrepaul Recently shifted teams and focus at Cypress and likely won't be actively/frequently contributing to this project due to his shift in focus.

You're very welcome! Thanks also for merging!

I look forward to working with you in future in that case! Thanks for the heads-up about @jaffrepaul. I've been in touch with him via Discord and lately I haven't been hearing much from him on that channel.

Comment (emilyrohrbough) [reply README.md:125]: I'm looking forward to working with you as well! You've made a large, GOOD, impression on our team and we have been super excited about all of the efforts and contributions you have made to help us get things on track and up-to-date! We are super super appreciative!

Comment (MikeMcC399) [reply README.md:125]: @emilyrohrbough 
Thanks very much! That's nice to hear 🙂 ! The action itself is in good shape. It's now more a question of getting documentation and tutorials, etc. up to date.
